staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -59,15 +59,89 @@ type ThreadSafeStore interface {
 	Resync() error
 }
 
+// storeIndex implements the indexing functionality for Store interface
+type storeIndex struct {
```

Comment:
Nit: the definition and methods of this type are kind of split as of now (some defined here and some towards the end of the file) - if possible, it'd be nice to have them together.